stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -872,8 +872,8 @@ public class ChatClient internal constructor(
     }
 
     @CheckResult
-    public fun deleteMessage(messageId: String): Call<Message> {
-        return api.deleteMessage(messageId)
+    public fun deleteMessage(messageId: String, hard: Boolean = false): Call<Message> {
```

Comment:
Sometimes @JvmOverloads produces lots of overloads and it feels messy to put it. But here I think it is fine. Only 1 overload to make the life of Java users easier.